torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -452,13 +482,13 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
       GRAPH_DEBUG("After guardDifferentiableGraph:\n", *copy);
       auto diff_graph = std::move(dnode->g(attr::Subgraph));
       Gradient gradient = differentiate(diff_graph);
+      removeProfileNodes(gradient.f);
       GRAPH_DEBUG("Forward graph:\n", *(gradient.f));
```

Comment:
Can we also be sure that we're removing specialized types here ? `removeTensorTypeSpecializations`